test/C/src/target/LinkLibrariesNoCMake.lf

```diff
@@ -0,0 +1,15 @@
+/**
+ * Test that the link-libraries target property is working correctly.
+ * The failure for this test is failure to compile.
+ */
+target C {
+    link-std-libs: "m",
```

Comment:
I am confused -- please forgive me. On my machine, this test compiles even if this line is deleted. It even compiles using a version of `lfc` that is not from this branch. What is this test supposed to verify?

(Incidentally, `LinkLibraries.lf` also compiles on my machine even if I delete the preamble, albeit with a warning. I also do not understand why `link-std-libs: "m"` is commented out there.)

Reply:
I commented out that line to verify whether it would only compile on Linux or also on other platforms. The test was contributed by @Soroosh129 without that line not commented out. But when I commented it out it still worked, so I'm trying to figure out what problem we're trying to fix here...